kubernetes-membership-scheme/src/main/java/org/wso2/carbon/membership/scheme/kubernetes/resolver/ApiBasedPodIpResolver.java

```diff
@@ -0,0 +1,241 @@
+        }
+
+        if (StringUtils.isEmpty(kubernetesMasterPassword)) {
+            kubernetesMasterPassword = getParameterValue(Constants.PARAMETER_NAME_KUBERNETES_API_SERVER_PASSWORD, "");
+        }
+
+        if (StringUtils.isEmpty(skipMasterVerificationValue)) {
+            skipMasterVerificationValue = getParameterValue(Constants.PARAMETER_NAME_KUBERNETES_MASTER_SKIP_SSL_VERIFICATION, "false");
+        }
+
+        skipMasterSSLVerification = Boolean.parseBoolean(skipMasterVerificationValue);
+
+        log.info(String.format("Kubernetes clustering configuration: [api-server] %s [namespace] %s [services] %s [skip-master-ssl-verification] %s",
+                kubernetesApiServerUrl, getKubernetesNamespace(), getKubernetesServices(), skipMasterSSLVerification));
+    }
+
+    @Override
+    public Set<String> resolveAddresses() throws KubernetesMembershipSchemeException {
+
+        final String apiContext = String.format(Constants.ENDPOINTS_API_CONTEXT, getKubernetesNamespace());
+        final Set<String> containerIPs = new HashSet<>();
+
+        for (String kubernetesService : getKubernetesServicesArray()) {
+            // Create k8s api endpoint URL
+            URL apiEndpointUrl = createUrl(kubernetesApiServerUrl, apiContext + kubernetesService.trim());
+
+            // Create http/https k8s api endpoint
+            KubernetesApiEndpoint apiEndpoint = createAPIEndpoint(apiEndpointUrl);
+
+            // Create the connection and read k8s service endpoints
+            Endpoints endpoints;
+            try {
+                endpoints = getEndpoints(connectAndRead
+                        (apiEndpoint, kubernetesMasterUsername, kubernetesMasterPassword));
+
+            } catch (IOException e) {
+                throw new KubernetesMembershipSchemeException("Could not get the Endpoints", e);
+
+            } finally {
+                apiEndpoint.disconnect();
+            }
+
+            if (endpoints != null) {
+                if (endpoints.getSubsets() != null && !endpoints.getSubsets().isEmpty()) {
+                    // Reading IP addresses from two lists
+                    log.info("Reading IP addresses from endpoints");
+                    for (Subset subset : endpoints.getSubsets()) {
+                        if (subset.getAddresses() != null) {
+                            for (Address address : subset.getAddresses()) {
+                                containerIPs.add(address.getIp());
+                            }
+                        }
+                        if (subset.getNotReadyAddresses() != null) {
+                            for (Address address : subset.getNotReadyAddresses()) {
+                                containerIPs.add(address.getIp());
+                            }
+                        }
+                    }
+                }
+            } else {
+                log.info("No endpoints found at " + apiEndpointUrl.toString());
```

Comment:
AFAIK we are not aborting server initialization due to no member IPs being available. However, IMO its better to make the log level WARN.